### PR TITLE
KEYCLOAK-14137 Upgrade Authentication selector screen

### PR DESCRIPTION
**Pull request by Pepo48**
* edited related css and ftl theme resources
* fixed affected tests

Signed-off-by: Peter Zaoral <pzaoral@redhat.com>

Cc: @mabartos @ssilvert @vmuzikar @drichtarik 
Just a quick note - @mabartos's KEYCLOAK-14139 should be merged first. There probably will be a conflict with the `stylesCommon` paths in `theme.properties` but I will fix it easily just by accepting his version.

### CODE REVIEW

File: themes/src/main/resources/theme/keycloak/login/theme.properties
Comment (mabartos) [inline themes/src/main/resources/theme/keycloak/login/theme.properties:5]: @Pepo48 Could you please remove the zocial css? There is no need for that anymore. It may avoid some merge conflicts.

**Review by mabartos — CHANGES_REQUESTED** [review]
@Pepo48 Thanks for PR, LGTM! Just a few minor changes would be good to do.

- `Zocial` styles (below in comment)
- Change `displayInfo` to false (It'd be better to change `displayInfo` property of the template to `false`. Gray info section is shown below the selectors, but there is no defined info message and that seems a little bit strange.)

File: themes/src/main/resources/theme/keycloak/login/theme.properties
Comment (Pepo48) [reply themes/src/main/resources/theme/keycloak/login/theme.properties]: @mabartos thanks for the review! Both issues should be fixed now.

**Review by mabartos — DISMISSED** [review]
@Pepo48 Thanks!

**Review by vmuzikar — CHANGES_REQUESTED** [review]
I haven't yet done a full review but there seem to be some compilation errors in Account Console tests:
https://keycloak-jenkins.com/job/universal-test-pipeline-server/2185/execution/node/458/log/

@Pepo48 Could you please fix them? Thanks!

**Review by vmuzikar — COMMENTED** [review]
Changes look good, nice work @Pepo48!

The compilation error doesn't seem to be related however I'd like to have it resolved before merging this in order to run the Account Console tests. Maybe could you please try to rebase? It's possible someone already fixed it.

One more thing. I'm a bit missing some hover effect when hovering the mouse cursor above the authenticators. :)

**Comment by mabartos** [comment]
> I haven't yet done a full review but there seem to be some compilation errors in Account Console tests:
> https://keycloak-jenkins.com/job/universal-test-pipeline-server/2185/execution/node/458/log/
> 

@Pepo48 It's caused by this [commit](https://github.com/keycloak/keycloak/commit/ff05072c164d95e5084fba7d556820d1ccbe7f31). You have to add next parameter (null) to [DeviceActivityTest](https://github.com/keycloak/keycloak/blob/4541a1b250186d9ffe1a7ac3e4dbae80c0a04bdc/testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/DeviceActivityTest.java#L269). Problem should be solved.

**Comment by mabartos** [comment]
> One more thing. I'm a bit missing some hover effect when hovering the mouse cursor above the authenticators. :)
+1 for hover effect :)

**Comment by Pepo48** [comment]
@vmuzikar @mabartos thanks for all the suggestions, Moreover, I found out that the css styling is a bit broken even on this screen:
obrázok
I will take a look into it after public holiday.

**Comment by Pepo48** [comment]
> 
> 
> > I haven't yet done a full review but there seem to be some compilation errors in Account Console tests:
> > https://keycloak-jenkins.com/job/universal-test-pipeline-server/2185/execution/node/458/log/
> 
> @Pepo48 It's caused by this [commit](https://github.com/keycloak/keycloak/commit/ff05072c164d95e5084fba7d556820d1ccbe7f31). You have to add next parameter (null) to [DeviceActivityTest](https://github.com/keycloak/keycloak/blob/4541a1b250186d9ffe1a7ac3e4dbae80c0a04bdc/testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/DeviceActivityTest.java#L269). Problem should be solved.

@mabartos Thanks! It's already fixed.

**Comment by Pepo48** [comment]
> One more thing. I'm a bit missing some hover effect when hovering the mouse cursor above the authenticators. :)
> +1 for hover effect :)


@vmuzikar @mabartos guys I just wanted to discuss briefly how exactly should the hover effect looks like?
So far I've come up with this:
obrázok
obrázok

**Comment by Pepo48** [comment]
> 
> 
> @vmuzikar @mabartos thanks for all the suggestions, Moreover, I found out that the css styling is a bit broken in IE11 even on this screen:
> obrázok
> I will take a look into it after public holiday.

@vmuzikar @mabartos meanwhile I also added the IE11 support (with the hover effect included):
obrázok

By implementing this I covered all the review requests mentioned above. Let me know in case of any further adjustments. Thanks!

**Comment by mabartos** [comment]
Old one:
oldAuth

New one:
newAuth

LGTM. But I have one minor thing. IMO, the font size is too large in the new Auth selection screen. I'd keep the same size like in the old one. WDYT @Pepo48 ?

**Comment by Pepo48** [comment]
> LGTM. But I have one minor thing. IMO, the font size is too large in the new Auth selection screen. I'd keep the same size like in the old one. And maybe the icons align to center? But these are only minor things, so if you think it's good, I'll approve that. WDYT @Pepo48 ?

What about these sizes?
image

**Comment by mabartos** [comment]
> > LGTM. But I have one minor thing. IMO, the font size is too large in the new Auth selection screen. I'd keep the same size like in the old one. And maybe the icons align to center? But these are only minor things, so if you think it's good, I'll approve that. WDYT @Pepo48 ?
> 
> What about these sizes?
> image

Much better. Font size should be similar like in all KC forms. Thanks :)

**Comment by Pepo48** [comment]
just for the sake of completeness I am attaching an IE11 variant:
obrázok

**Review by mabartos — DISMISSED** [review]
Well done! Thanks @Pepo48 !

**Comment by Pepo48** [comment]
I'm attaching the UI tests result: 
https://keycloak-jenkins.com/job/universal-test-pipeline-server/2260/

File: themes/src/main/resources/theme/keycloak/login/resources/css/login.css
Comment (vmuzikar) [inline themes/src/main/resources/theme/keycloak/login/resources/css/login.css:577]: I'm afraid I have to bug you a bit more with IE. It seems there's some overflow when the text is longer.
Capture

File: themes/src/main/resources/theme/keycloak/login/resources/css/login.css
Comment (vmuzikar) [inline themes/src/main/resources/theme/keycloak/login/resources/css/login.css:636]: This is very subjective (as usually with UI stuff ;)) and it's definitely not a blocker but I think it looks a bit awkward that the blue border is just on top. Maybe it'd look better if it would be at the bottom as well? I don't know... Is there any PF4 best practice on this?

**Review by vmuzikar — COMMENTED** [review]
Thanks for the update, @Pepo48!

Added two comments. Also, I think we need to run the base testsuite in the testpipeline as Travis seems to be stuck and won't be working anymore.

File: themes/src/main/resources/theme/keycloak/login/resources/css/login.css
Comment (Pepo48) [reply themes/src/main/resources/theme/keycloak/login/resources/css/login.css:636]: @vmuzikar: Martin asked me the exact same question couple days ago. I remember that during the initial polishing in May I removed the horizontal bottom line based on the comment[1] from @stianst but I left the top horizontal line based on PF guidelines[2]:
image

So I don't know now, any thoughts?

[1] https://issues.redhat.com/browse/KEYCLOAK-12824?focusedCommentId=14084158&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14084158
[2] https://www.patternfly.org/v4/components/login-page/design-guidelines/

**Comment by Pepo48** [comment]
I'm attaching the base tests result (without failures): 
https://keycloak-jenkins.com/job/universal-test-pipeline-server/2284/

File: themes/src/main/resources/theme/keycloak/login/resources/css/login.css
Comment (vmuzikar) [reply themes/src/main/resources/theme/keycloak/login/resources/css/login.css:636]: I wouldn't add the bottom line to normal (non-hover) state to keep it compliant with the [PF4 guidelines](https://www.patternfly.org/v4/images/multi-factor.1cc14528bafc7a0cac690fb3105ee2e1.png).

So maybe we could take some other approach for the hover effect (which is not covered in the guideline if I'm correct). What about adding a light blue background color to the button on hover (instead of changing the line and text color)? WDYT @mabartos @Pepo48?

File: themes/src/main/resources/theme/keycloak/login/resources/css/login.css
Comment (Pepo48) [reply themes/src/main/resources/theme/keycloak/login/resources/css/login.css:577]: Fixed by adding:
https://github.com/Pepo48/keycloak/blob/4a2849b1467b8567a48328a776acef004b568e6e/themes/src/main/resources/theme/keycloak/login/resources/css/login.css#L605-L608

It just seems to be another IE specific bug/feature, where it needs the explicit request[1].
[1] https://stackoverflow.com/a/35113633

File: themes/src/main/resources/theme/keycloak/login/resources/css/login.css
Comment (Pepo48) [reply themes/src/main/resources/theme/keycloak/login/resources/css/login.css]: @vmuzikar (Cc: @mabartos) what about this approach?
image
I pushed it here in a separate commit to make the changes more visible, I will squash it once we agree on the final design :)
https://github.com/keycloak/keycloak/pull/7488/commits/ccca13f8d6331a59829fdf038ea29f8a196ea98a#diff-cc2b625f843865ee5186a5b77258a0b2fb66c0754b9e7e2be0bd0247408015e2

Comment (Pepo48) [reply themes/src/main/resources/theme/keycloak/login/resources/css/login.css]: @vmuzikar @mabartos Haley from PF team suggested that we might to take inspiration from the hover effect of selectable-tabs PF component [1]. But I guess it's up to us to make the final decision :)

[1] https://www.patternfly.org/v4/components/data-list/#selectable-rows

Comment (vmuzikar) [reply themes/src/main/resources/theme/keycloak/login/resources/css/login.css]: @Pepo48 I'm afraid it's not entirely up to us. :( We need to stick whatever PF suggests.

**Review by mabartos — DISMISSED** [review]
LGTM! @Pepo48 Thanks!

**Review by vmuzikar — DISMISSED** [review]
After discussing the hover effect with @Pepo48, I think we're good to go.

PF currently doesn't have a component for this specific case, so IMHO it's ok to improvise here a bit. It looks nice, it's not a deviation in PF style, so why not. :)

Nice work, thanks @Pepo48!

**Comment by Pepo48** [comment]
Squashed, I ran the tests, let's wait for the results.

**Comment by Pepo48** [comment]
I'm attaching the results once again:
* base tests:
https://keycloak-jenkins.com/job/universal-test-pipeline-server/2345/
* UI tests:
https://keycloak-jenkins.com/job/universal-test-pipeline-server/2341/

**Review by vmuzikar — APPROVED** [review]

**Review by mposolda — APPROVED** [review]
@Pepo48 @vmuzikar @mabartos Thanks everyone!

**Pull request merged**
